source/coap_connection_handler.c

```diff
@@ -793,9 +801,13 @@ coap_conn_handler_t *connection_handler_create(receive_from_socket_cb *recv_from
 
     return handler;
 }
-void connection_handler_destroy(coap_conn_handler_t *handler)
+
+void connection_handler_destroy(coap_conn_handler_t *handler, bool multicast_group_leave)
 {
     if(handler){
+        if (multicast_group_leave) {
```

Comment:
(Indeed the only real reason for the join being associated with a socket is to provide that automatic clean-up - it doesn't directly affect reception behaviour on a per-socket basis. A socket would still receive packets to ff05::fd if it hadn't joined but another socket had.)